test/unit/org/apache/cassandra/schema/createlike/CreateLikeTest.java

```diff
@@ -576,28 +581,135 @@ public void testTriggerOperationOnCopiedTable()
+    {
+        String sourceTb = createTable(sourceKs, "CREATE TABLE %s (a int PRIMARY KEY, b int, c text, d int)", "sourcetb");
+        createIndex(sourceKs, "CREATE INDEX ON %s (b)");
+        createTrigger(sourceKs, "sourcetb", "trigger_1", "CREATE TRIGGER %s ON %s USING '" + triggerClass1 + "'");
+        String targetTbWithAll = createTableLike("CREATE TABLE %s LIKE %s WITH ALL AND crc_check_chance = 0.8 AND cdc = true", sourceTb, sourceKs, targetKs);
+        assertTableMetaEqualsWithoutKs(sourceKs, targetKs, sourceTb, targetTbWithAll, false, false, false);
+
+        TableMetadata source = getTableMetadata(sourceKs, sourceTb);
+        TableMetadata target = getTableMetadata(targetKs, targetTbWithAll);
+        assertNotEquals(source.params, target.params);
+        assertTrue(target.params.cdc);
+        assertFalse(source.params.cdc);
+        assertEquals(0.8, target.params.crcCheckChance, 0.0);
+
+        String targetTbWithTrigger = createTableLike("CREATE TABLE %s LIKE %s WITH TRIGGER AND crc_check_chance = 0.8 AND cdc = true", sourceTb, sourceKs, targetKs);
+        assertTableMetaEqualsWithoutKs(sourceKs, targetKs, sourceTb, targetTbWithTrigger, false, false, false);
+
+        source = getTableMetadata(sourceKs, sourceTb);
+        target = getTableMetadata(targetKs, targetTbWithTrigger);
+        assertNotEquals(source.params, target.params);
+        assertTrue(target.params.cdc);
+        assertFalse(source.params.cdc);
+        assertEquals(0.8, target.params.crcCheckChance, 0.0);
+    }
+
+    @Test
+    public void testUnSupportedCopyofIndexAndTrigger() throws Throwable
+    {
+        createTable(sourceKs, "CREATE TABLE %s (a int PRIMARY KEY, b int, c text, d int)", "sourcetb");
+        createIndex(sourceKs, "CREATE INDEX ON %s (b)");
+        createTrigger(sourceKs, "sourcetb", "trigger_1", "CREATE TRIGGER %s ON %s USING '" + triggerClass1 + "'");
+        // ALL and TRIGGERS is not allowed
+        assertInvalidThrowMessage("It is not allowed to use ALL option together with [TRIGGER]",
+                                  InvalidRequestException.class,
+                                  "CREATE TABLE " + targetKs + ".targettb LIKE " + sourceKs + ".sourcetb WITH TRIGGER AND ALL");
+        // ALL and INDEXES is not allowed
+        assertInvalidThrowMessage("It is not allowed to use ALL option together with [INDEX]",
+                                  InvalidRequestException.class,
+                                  "CREATE TABLE " + targetKs + ".targettb LIKE " + sourceKs + ".sourcetb WITH INDEX AND ALL");
+        // ALL and INDEXES and TRIGGERS is not allowed
+        assertInvalidThrowMessage("It is not allowed to use ALL option together with [INDEX, TRIGGER]",
+                                  InvalidRequestException.class,
+                                  "CREATE TABLE " + targetKs + ".targettb LIKE " + sourceKs + ".sourcetb WITH INDEX AND TRIGGER AND ALL");
+    }
+
```

Comment:
What about, for completion, adding tests for tables without indexes and triggers that are being copied with `LIKE foo WITH INDEX/TRIGGER/ALL`?